### DISCUSSION TITLE
What are the Python versions that each version of Redash depends on?

### DISCUSSION THREAD

**Discussion by Jengro777** [RedashCommunity/redash#23] [General]
What are the Python versions that each version of Redash depends on?

##
"Please record the versions of Python dependent packages here."
```
#psycopg2==2.8.3 #no
psycopg2==2.9.6

#greenlet==0.4.16 #no
greenlet==2.0.2

#pystache==0.5.4 #no
pystache==0.6.0 

#gevent==1.4.0 #no
gevent==22.10.2

**Comment by justinclift** [comment]
I've been doing stuff on Ubuntu 20.04 LTS, which comes with Python 3.8 by default.

**Reply by Jengro777** [reply]
Redash cannot be installed with the gevent dependency version 1.4.0 on Python 3.9.2 on Debian 11.

**Comment by Jengro777** [comment]
ERROR: Could not find a version that satisfies the requirement gspread==3.1.0
ERROR: No matching distribution found for gspread==3.1.0

**Comment by Jengro777** [comment]
.env.example

Where is this file? It seems to be missing in the project.

**Reply by gaecoli** [reply]
> .env.example
> 
> Where is this file? It seems to be missing in the project.

you can look up [redash doc](https://redash.io/help/open-source/admin-guide/env-vars-settings).

**Reply by justinclift** [reply]
As a data point, the `env` file (and variants like `.env`) get created by the setup script:

https://github.com/getnewdash/setup/blob/f46decba85e20408e43c311348ce2ce715297379/setup.sh#L36-L48
